src/io/flutter/vmService/DisplayRefreshRateManager.java

```diff
@@ -80,18 +80,18 @@ public Double getCurrentDisplayRefreshRateRaw() {
   private CompletableFuture<Double> getDisplayRefreshRate() {
     final double unknownRefreshRate = 0.0;
 
-    final String flutterViewId = vmServiceManager.getFlutterViewId().exceptionally(exception -> {
-      // We often see "java.lang.RuntimeException: Method not found" from here; perhaps a race condition?
-      LOG.warn(exception.getMessage());
-      return null;
-    }).join();
-
-    if (flutterViewId == null) {
+    try {
+      final String flutterViewId = vmServiceManager.getFlutterViewId().exceptionally(exception -> {
+        // We often see "java.lang.RuntimeException: Method not found" from here; perhaps a race condition?
+        LOG.warn(exception.getMessage());
+        return null;
+      }).get();
```

Comment:
It looks like `get()` may also block the thread until the value is ready?

I believe the underlying issue here is that we're using clocking calls and this code path can be called from the event thread. I suspect what we need to do is to move to a non-blocking call (which may be something like `.whenComplete()`?), and then use that to determine the display refresh rate, and return the value as a future.

(Java futures definitely have a much larger API surface area than the Dart variant)

Reply:
The issue with `whenComplete` is that it returns a `CompletableFuture<T> `. In this case T is a String because that is the value we expect from `getFlutterViewId`. However, we need to return CompletableFuture<double> for displayRefreshRate.

Reply:
You may try creating a `CompletableFuture<Double> foo` at the start of the method, starting the future work to get the flutterViewId, and return the `foo` completable future. When you have the flutter view ID, you can then request the display rate. Once you have a value for that you should be able to call `foo.complete(displayRefreshRate)`.

Reply:
I think I may have found a way around this by returning outside of whenComplete. PTAL.